pkg/fqdn/name_manager_bench_test.go

```diff
@@ -78,3 +79,43 @@ func BenchmarkUpdateGenerateDNS(b *testing.B) {
 			}})
 	}
 }
+
+// BenchmarkFqdnCache tests how slow a full dump of DNSHistory from a number of
+// endpoints is. Each endpoints has 1000 DNS lookups, each with 10 IPs. The
+// dump iterates over all endpoints, lookups, and IPs.
+func BenchmarkFqdnCache(b *testing.B) {
```

Comment:
Not strictly related to this PR, since the bench logic is the same as before, but I don't think this benchmark is correct.

I think there are two errors:

1) We shouldn't change the input size from one call to another, as this will make not possible for the Go framework to find a stable value for the elapsed time measurement. Instead we should choose a fixed number of endpoints (I tried some values and 8 endpoints seems enough).
2) The code to be measured should be executed `b.N` times at each benchmark call, but here `GetDNSHistoryModel` is executed only once. Again, this can interfere with the framework trying to find a stable value for the elapsed time.

From the Go testing package [docs](https://pkg.go.dev/testing#hdr-Benchmarks):

```
The benchmark function must run the target code b.N times. During benchmark execution, b.N is adjusted until the benchmark function lasts long enough to be timed reliably.
```

So, we should change the benchmark this way:

```
	const endpoints = 8

	caches := make([]*DNSCache, 0, endpoints)
	for i := 0; i < endpoints; i++ {
	
	... <init as before> ...
	
	b.ResetTimer()

	prefixMatcher := func(_ netip.Addr) bool { return true }
	nameMatcher := func(_ string) bool { return true }

	for i := 0; i < b.N; i++ {
		nameManager.GetDNSHistoryModel("", prefixMatcher, nameMatcher, "")
	}
```

Reply:
Thanks for the detail review, what you describe makes sense. I filed https://github.com/cilium/cilium/pull/32276 as a follow-up PR to fix this.